lib/active_record/connection_adapters/postgis/postgis_database_tasks.rb

```diff
@@ -36,54 +36,54 @@ def create(master_established = false)
 
         # Override to use su_username and su_password
         def establish_master_connection
-          establish_connection(configuration.merge(
-            "database"           => "postgres",
-            "password"           => su_password,
-            "schema_search_path" => "public",
-            "username"           => su_username
+          establish_connection(configuration_hash.merge(
+            database:           "postgres",
+            password:           su_password,
+            schema_search_path: "public",
+            username:           su_username
           ))
         end
 
         def establish_su_connection
-          establish_connection(configuration.merge(
-            "password"           => su_password,
-            "schema_search_path" => "public",
-            "username"           => su_username
+          establish_connection(configuration_hash.merge(
+            password:           su_password,
+            schema_search_path: "public",
+            username:           su_username
           ))
         end
 
         def username
-          @username ||= configuration["username"]
+          @username ||= configuration_hash[:username]
         end
 
         def quoted_username
           @quoted_username ||= ::ActiveRecord::Base.connection.quote_column_name(username)
         end
 
         def password
-          @password ||= configuration["password"]
+          @password ||= configuration_hash[:password]
         end
 
         def su_username
-          @su_username ||= configuration["su_username"] || username
+          @su_username ||= configuration_hash[:su_username] || username
```

Comment:
A more correct pattern for using default values in ruby is
```suggestion
          @su_username ||= configuration_hash.fetch(:su_username, username)
```

This is more resilient to possible falsy values, and in terms of code readability, is more expressive IMHO. Could do the change here and for other occurences ?

Reply:
Done for both `su_username` and `su_password`. I can't see other occurrences where this apply in this file